torch_geometric/datasets/infection_dataset.py

```diff
@@ -0,0 +1,163 @@
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import torch
+
+from torch_geometric.data import Data, InMemoryDataset
+from torch_geometric.datasets.graph_generator import GraphGenerator
+from torch_geometric.explain import Explanation
+from torch_geometric.seed import seed_everything
+from torch_geometric.utils import k_hop_subgraph, sort_edge_index, to_networkx
+
+
+class InfectionDataset(InMemoryDataset):
+    r"""
+    Generates a synthetic infection dataset for evaluating explainabilty
+    algorithms, as described in the `"Explainability Techniques for Graph
+    Convolutional Networks" <https://arxiv.org/abs/1905.13686>`__ paper.
+    The :class:`~torch_geometric.datasets.InfectionDataset` creates synthetic
+    graphs coming from a
+    :class:`~torch_geometric.datasets.graph_generator.GraphGenerator` with
+    :obj:`num_infected` randomly assigned infected nodes. The dataset then
+    creates a node classification task of predicting length of the shortest
+    path from infected nodes with ground-truth node-level mask and explanation
+    paths.
+
+    For example, to generate a random Erdos-Renyi (ER) infection graph
+    with 500 nodes and 0.004 probability of edge, write:
+
+    .. code-block:: python
+
+        from torch_geometric.datasets import InfectionDataset
+        from torch_geometric.datasets.graph_generator import ERGraph
+
+        dataset = InfectionDataset(
+            graph_generator=ERGraph(num_nodes=500, edge_prob=0.004),
+        )
+
+    Args:
+        graph_generator (GraphGenerator or str): The graph generator to be
+            used, *e.g.*,
+            :class:`torch.geometric.datasets.graph_generator.BAGraph`
+            (or any string that automatically resolves to it).
+        num_graphs (int, optional): The number of graphs to generate.
+            (default: :obj:`1`)
+        graph_generator_kwargs (Dict[str, Any], optional): Arguments passed to
+            the respective graph generator module in case it gets automatically
+            resolved. (default: :obj:`None`)
+        seeds (List[int], optional): List of seeds to generate the infection
+            graphs. (default: :obj:`None`)
```

Comment:
```suggestion
        seeds (List[int], optional): List of seeds. Each seed is used to generate one
            infection graph. (default: :obj:`None`)
```
or adding something like:  `The number of seeds needs to be the same as the number of graphs`